L1Trigger/L1THGCal/interface/HGCalCoarseTriggerCellMapping.h

```diff
@@ -0,0 +1,77 @@
+#ifndef __L1Trigger_L1THGCal_HGCalCoarseTriggerCellMapping_h__
+#define __L1Trigger_L1THGCal_HGCalCoarseTriggerCellMapping_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetIdToROC.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+      
+class HGCalCoarseTriggerCellMapping
+{
+  public:
+  HGCalCoarseTriggerCellMapping(const std::vector<unsigned>& ctcSize);
+    uint32_t getEvenDetId(uint32_t tcid) const;
+    std::vector<uint32_t> getConstituentTriggerCells( uint32_t ctcId ) const;
+    GlobalPoint getCoarseTriggerCellPosition( uint32_t tcid ) const;
+    uint32_t getCoarseTriggerCellId(uint32_t detid) const ;
+    void checkSizeValidity(int ctcSize)const;
+    void eventSetup(const edm::EventSetup& es) {triggerTools_.eventSetup(es);}
+
+    static constexpr int kCTCsizeCoarse_ = 16;
+    static constexpr int kCTCsizeMid_ = 8;
+    static constexpr int kCTCsizeFine_ = 4;
+    static constexpr int kCTCsizeVeryFine_ = 2;    
+    static constexpr int kCTCsizeIndividual_ = 1;    
+
+  private:
+
+
+    static const std::map<int,int> kSplit_;
+    static const std::map<int,int> kSplit_v9_;
+    static const std::map<int,int> kSplit_v9_Scin_;
+    //    static constexpr int kWafer_offset_ = 6;
+    static constexpr int kSTCidMask_ = 0x3f;
+    static constexpr int kSTCidMask_v9_ = 0xf;
+    static constexpr int kSplit_v8_Coarse_ = 0x30;
+    static constexpr int kSplit_v8_Mid_ = 0x38;
+    static constexpr int kSplit_v8_Fine_ = 0x3a;
+    static constexpr int kSplit_v8_VeryFine_ = 0x3e;
+    static constexpr int kSplit_v8_Individual_ = 0x3f;
+    //    static constexpr int kNLayers_ = 3;
+    static constexpr int kNLayers_ = 4;
+    static constexpr int kSplit_v9_Coarse_ = 0;
+    static constexpr int kSplit_v9_Mid_ = 0x2;
+    static constexpr int kSplit_v9_Fine_ = 0xa;
+    static constexpr int kSplit_v9_VeryFine_ = 0xb;
+    static constexpr int kSplit_v9_Individual_ = 0xf;
+
+    static constexpr int kSplit_v9_Scin_Coarse_ = 0x1f9fc;
+    static constexpr int kSplit_v9_Scin_Mid_ = 0x1fdfc;
+    static constexpr int kSplit_v9_Scin_Fine_ = 0x1fdfe;
+    static constexpr int kSplit_v9_Scin_VeryFine_ = 0x1fffe;
+    static constexpr int kSplit_v9_Scin_Individual_ = 0x1ffff;
+
+    //For coarse TCs
+    static constexpr int kRocShift_ = 4;
+    //    static constexpr int kRocMask_ = 0xC0;
+    static constexpr int kRocMask_ = 0xf0;
+    static constexpr int kRotate4_ = 4;
+    static constexpr int kRotate7_ = 7;
+    static constexpr int kUShift_ = 2;
+    static constexpr int kVShift_ = 0;
+    static constexpr int kUMask_ = 0xc;
+    static constexpr int kVMask_ = 0x3;
+    static constexpr int kHGCalCellUOffset_      = 0;
```

Comment:
The U,V in the superTC definition are actually U' and V' taking 2 bits each with the additional bits being taken by the ROC number. Maybe it would make more sense to call it `kUprimeMask` or similar?

Reply:
`kHGCalCellUMask_` is 4 bits in the current definition below, not 2?

Reply:
yes this is the mask to get back to "normal" U after receiving a U prime.